gdx/test/com/badlogic/gdx/utils/LongArrayTest.java

```diff
@@ -0,0 +1,425 @@
+/*
+ * Copyright 2024 See AUTHORS file
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.badlogic.gdx.utils;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class LongArrayTest {
+	/** Test of the different adding methods */
+	@Test
+	public void addTest () {
+		/*
+		 * Test of the classic add(long value) method, it should be adding a number at the first available place.
+		 */
+		LongArray longArray1 = new LongArray(3);
+		longArray1.add(3);
+		Assert.assertArrayEquals(new long[] {3}, longArray1.toArray());
+
+		/*
+		 * Test of the add(long value1,long value2) method, it should be adding these numbers in the array, when it is possible.
+		 */
+		LongArray longArray2 = new LongArray();
+		longArray2.add(1, 2);
+		Assert.assertArrayEquals(new long[] {1, 2}, longArray2.toArray());
+
+		/*
+		 * Some test with the addAll(LongArray array) and addAll (long... array) methods. The second call should resize longArray3
+		 * to a size of 17.
+		 */
+		LongArray longArray3 = new LongArray();
+		longArray3.addAll(longArray2);
+		Assert.assertArrayEquals(longArray2.toArray(), longArray3.toArray());
+		longArray3.addAll(longArray1);
```

Comment:
I find this quite difficult to read and relying on the results of the previous tests weakens this. I would include `{3}` and `{1, 2}` directly in this test. obigu might agree? https://github.com/libgdx/libgdx/pull/7518#discussion_r1851716120